backends-velox/src/main/scala/org/apache/spark/sql/execution/VeloxColumnarToRowExec.scala

```diff
@@ -134,6 +135,8 @@ class ColumnarToRowRDD(
             if (!hasNext && !closed) {
               jniWrapper.nativeClose(c2rId)
               closed = true
+
+              TaskResources.releaseCurrentResources()
```

Comment:
Would you like to explain a bit about this? Thanks!

Reply:
If there is a case: (G for Gluten, V for Vanilla Spark)

```
G_ShuffleRead -> C2R_1 -> V_Plan_1 -> R2C_1 -> G_Plan_1 -> C2R_2 -> V_ShuffleWrite
```

Once C2R_1 reached EoS, C2R_2 might still consume data. Is the case covered by the code?

Reply:
Sure, the key changes is `RESOURCE_REGISTRIES` which contains mapping of 
(TaskContext -> (ThreadID -> TaskResourceRegistry)) now, to handle different thread computation under same Spark TaskContext scope, let it could be released.

If all computations are offloaded to native, there should no multi computation under same Spark task scope. If we fallback some special operators, c2r will be inserted, and this parts of computation could happens in another thread, such as PythonRunner, so we need add extra mapping to release resources created in another thread.

Reply:
> Once C2R_1 reached EoS, C2R_2 might still consume data. Is the case covered by the code?

Yes, I add a UnsafeRow#copy at the end of ColumnarToRowRDD.